torch/csrc/autograd/variable.h

```diff
@@ -525,7 +574,9 @@ inline Variable make_variable(at::Tensor data, Edge gradient_edge) {
       !data.is_variable(),
       "Must not create a new variable from a variable, use its .data()");
   if (data.defined()) {
-    return Variable(c10::make_intrusive<Variable::Impl>(data, false, std::move(gradient_edge)));
+    auto data_copy = at::Tensor(data.getIntrusivePtr()->shallow_copy_and_detach());
+    auto autograd_meta = c10::guts::make_unique<Variable::AutogradMeta>();
+    return Variable(c10::make_intrusive<Variable::Impl>(data_copy, std::move(autograd_meta), false, std::move(gradient_edge)));
```

Comment:
why do you need to copy and why don't you mark the underlying data as not changeable?

Reply:
is it hard to identify the places where we need to delay setting `allow_tensor_metadata_change_`?  I don't quite understand where we need to do this and how hard it would be to change this.

Reply:
The heuristics is "we should only set this flag to false if we are in a *user-facing* API that returns a tensor whose TensorImpl is a shallow copy of another tensor's TensorImpl", and currently only Python `tensor.data` and Python/C++ `tensor.detach()` exhibit this behavior.

(I just added this to the comments for `allow_tensor_metadata_change_` in TensorImpl.h.)

Reply:
(Note to self: look at whether we can pass a parameter to `make_variable()` to control this flag)